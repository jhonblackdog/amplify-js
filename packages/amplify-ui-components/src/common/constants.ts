// Namespace
const AMPLIFY_UI = 'amplify-ui';

// Classnames
export const AMPLIFY_UI_BUTTON = `${AMPLIFY_UI}-button`;
export const AMPLIFY_UI_SECTION = `${AMPLIFY_UI}-section`;
export const AMPLIFY_UI_SECTION_HEADER = `${AMPLIFY_UI}-section-header`;
export const AMPLIFY_UI_HINT = `${AMPLIFY_UI}-hint`;
export const AMPLIFY_UI_LABEL = `${AMPLIFY_UI}-label`;
export const AMPLIFY_UI_LINK = `${AMPLIFY_UI}-link`;
export const AMPLIFY_UI_FORM_FIELD = `${AMPLIFY_UI}-form-field`;
export const AMPLIFY_UI_RADIO_BUTTON = `${AMPLIFY_UI}-radio-button`;
export const AMPLIFY_UI_CHECKBOX = `${AMPLIFY_UI}-checkbox`;
export const AMPLIFY_UI_TEXT_INPUT = `${AMPLIFY_UI}-text-input`;
export const AMPLIFY_UI_SCENE = `${AMPLIFY_UI}-scene`;
export const AMPLIFY_UI_SCENE_LOADING = `${AMPLIFY_UI}-scene-loading`;
export const AMPLIFY_UI_SELECT = `${AMPLIFY_UI}-select`;