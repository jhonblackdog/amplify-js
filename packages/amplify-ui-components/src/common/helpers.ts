import { cx } from 'emotion';

// TODO: rename
export const styleNuker = (override: boolean, cn: string, ecn: string): string => override ? cn : cx(ecn, cn);
