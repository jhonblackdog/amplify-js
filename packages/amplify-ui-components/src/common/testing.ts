export const pixelThreshold = 10;
