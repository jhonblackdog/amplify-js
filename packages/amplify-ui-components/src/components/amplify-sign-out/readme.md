# amplify-sign-out

<!-- Auto Generated Below -->


## Properties

| Property                | Attribute     | Description                                  | Type                                                | Default                           |
| ----------------------- | ------------- | -------------------------------------------- | --------------------------------------------------- | --------------------------------- |
| `buttonText`            | `button-text` | Text inside of the Sign Out button           | `string`                                            | `I18n.get(Translations.SIGN_OUT)` |
| `handleAuthStateChange` | --            | Auth state change handler for this component | `(nextAuthState: AuthState, data?: object) => void` | `dispatchAuthStateChangeEvent`    |


## Dependencies

### Used by

 - [amplify-greetings](../amplify-greetings)

### Depends on

- [amplify-button](../amplify-button)

### Graph
```mermaid
graph TD;
  amplify-sign-out --> amplify-button
  amplify-greetings --> amplify-sign-out
  style amplify-sign-out fill:#f9f,stroke:#333,stroke-width:4px
```

----------------------------------------------

*Built with [StencilJS](https://stenciljs.com/)*
