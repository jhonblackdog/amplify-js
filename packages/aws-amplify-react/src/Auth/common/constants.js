export const Constants = {
    AUTH_SOURCE_KEY: 'amplify-react-auth-source',
    AUTH0: 'auth0',
    GOOGLE: 'google',
    FACEBOOK: 'facebook',
    AMAZON: 'amazon',
    SIGN_IN_WITH_HOSTEDUI_KEY: 'amplify-react-signin-with-hostedUI'
};
