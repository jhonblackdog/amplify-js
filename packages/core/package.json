{
  "name": "@aws-amplify/core",
  "version": "1.0.26-unstable.1",
  "description": "Core category of aws-amplify",
  "main": "./dist/cjs/index.js",
  "module": "./dist/esm/src/index.js",
  "typings": "./dist/esm/src/index.d.ts",
  "sideEffects": false,
  "publishConfig": {
    "access": "public"
  },
  "scripts": {
    "test": "tslint 'src/**/*.ts' && jest --coverage",
    "build-with-test": "npm test && npm run build",
    "build:cjs": "tsc -p tsconfig.es5.json && rollup -c",
    "build:esm": "tsc -p tsconfig.es2015.json",
    "build": "npm run clean && npm run build:cjs && npm run build:esm",
    "clean": "rimraf tsc-out dist",
    "format": "tsfmt --useTsfmt tsfmt.json -r src/**/*.ts",
    "lint": "tslint 'src/**/*.ts'"
  },
  "react-native": {
    "./lib/ClientDevice": "./lib/ClientDevice/reactnative.js",
    "./lib/RNComponents": "./lib/RNComponents/reactnative.js",
    "./lib/StorageHelper": "./lib/StorageHelper/reactnative.js"
  },
  "repository": {
    "type": "git",
    "url": "git+https://github.com/aws/aws-amplify.git"
  },
  "author": "Amazon Web Services",
  "license": "Apache-2.0",
  "bugs": {
    "url": "https://github.com/aws/aws-amplify/issues"
  },
  "homepage": "https://github.com/aws/aws-amplify#readme",
  "devDependencies": {
    "find": "^0.2.7",
    "prepend-file": "^1.3.1",
    "prettier": "^1.7.4",
    "typescript": "^3.2.2"
  },
  "dependencies": {
    "aws-sdk": "2.329.0",
    "url": "^0.11.0"
  },
  "jest": {
    "transform": {
      "^.+\\.(js|jsx|ts|tsx)$": "<rootDir>../../node_modules/ts-jest/preprocessor.js"
    },
    "testRegex": "(/__tests__/.*|\\.(test|spec))\\.(tsx?|jsx?)$",
    "moduleFileExtensions": [
      "ts",
      "tsx",
      "js",
      "json",
      "jsx"
    ],
    "testEnvironment": "jsdom",
    "mapCoverage": true,
    "coverageThreshold": {
      "global": {
        "branches": 0,
        "functions": 0,
        "lines": 0,
        "statements": 0
      }
    },
    "testURL": "http://localhost/",
    "coveragePathIgnorePatterns": [
      "/node_modules/"
    ]
  }
}
