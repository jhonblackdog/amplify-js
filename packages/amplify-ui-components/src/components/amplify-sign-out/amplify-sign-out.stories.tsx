import { storiesOf } from '@storybook/html';
import { knobs } from '../../common/testing';

const signOutStories = storiesOf('amplify-sign-out', module);

signOutStories.add('default', () => {
  const override = knobs.overrideStyleKnob();
  return `<amplify-sign-out override-style=${override}></amplify-sign-out>`;
});
