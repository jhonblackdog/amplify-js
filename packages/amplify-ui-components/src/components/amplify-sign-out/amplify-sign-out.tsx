import { Component, Prop, h } from '@stencil/core';
import { SIGN_OUT, NO_AUTH_MODULE_FOUND } from '../../common/constants';
import { AuthState, AuthStateHandler } from '../../common/types/auth-types';
import { Auth } from '@aws-amplify/auth';
import { dispatchToastHubEvent, dispatchAuthStateChangeEvent } from '../../common/helpers';

@Component({
  tag: 'amplify-sign-out',
  shadow: true,
})
export class AmplifySignOut {
  /** Passed from the Authenticator component in order to change Authentication state */
  @Prop() handleAuthStateChange: AuthStateHandler = dispatchAuthStateChangeEvent;
  /** Text inside of the Sign Out button */
  @Prop() buttonText: string = SIGN_OUT;

  async signOut(event) {
    if (event) event.preventDefault();

    // TODO: Federated Sign Out

    if (!Auth || typeof Auth.signOut !== 'function') {
      throw new Error(NO_AUTH_MODULE_FOUND);
    }

    try {
      await Auth.signOut();
      this.handleAuthStateChange(AuthState.SignedOut);
    } catch (error) {
      dispatchToastHubEvent(error);
    }
  }

  render() {
    return (
      <amplify-button onClick={event => this.signOut(event)} data-test="sign-out-button">
        {this.buttonText}
      </amplify-button>
    );
  }
}
