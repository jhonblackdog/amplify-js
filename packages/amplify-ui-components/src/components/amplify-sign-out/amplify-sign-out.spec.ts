import { newSpecPage } from '@stencil/core/testing';
import { AmplifySignOut } from './amplify-sign-out';
import { SIGN_OUT } from '../../common/constants';

describe('amplify-sign-out spec:', () => {
  describe('Component logic ->', () => {
    let signOut;

    beforeEach(() => {
      signOut = new AmplifySignOut();
    });

    it('should render default `buttonText`', () => {
      expect(signOut.buttonText).toEqual(SIGN_OUT);
    });

    it('should render `overrideStyle` to false by default', () => {
      expect(signOut.overrideStyle).toBe(false);
    });
  });
  describe('Render logic ->', () => {
    it('should render a `sign out` button by default', async () => {
      const page = await newSpecPage({
        components: [AmplifySignOut],
        html: `<amplify-sign-out></amplify-sign-out>`,
      });

      expect(page.root).toMatchSnapshot();
    });
  });
});
