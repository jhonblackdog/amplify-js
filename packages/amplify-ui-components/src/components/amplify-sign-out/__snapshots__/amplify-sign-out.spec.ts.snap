// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`amplify-sign-out spec: Render logic -> should render a \`sign out\` button by default 1`] = `
<amplify-sign-out>
  <amplify-button data-test="sign-out-button">
    Sign Out
  </amplify-button>
</amplify-sign-out>
`;
